sonic_platform_base/sensor_fs.py

```diff
@@ -0,0 +1,159 @@
+
+        for k,v in self.__defaults.items():
+            setattr(self, k, kw.get(k,v))
+
+        if (len(self.high_thresholds) != 3 or len(self.low_thresholds) != 3):
+            raise Exception('{}: Missing sensor thresholds'.format(self.name))
+
+        self.minimum_sensor = self.get_value()
+        self.maximum_sensor = self.minimum_sensor
+
+    def get_name(self):
+        """Returns the sensor name"""
+        return self.name
+
+    def get_value(self):
+        """Returns the sensor measurement"""
+        try:
+            with open(self.sensor) as f:
+                return int(f.readline().rstrip())
+        except:
+            return None
+
+    def get_high_threshold(self):
+        """Returns the sensor high threshold value"""
+        return self.high_thresholds[1]
+
+    def get_low_threshold(self):
+        """Returns the sensor low threshold value"""
+        return self.low_thresholds[1]
+
+    def set_high_threshold(self, value):
+        """Sets the sensor high threshold value"""
+        self.high_thresholds[1] = value
+        return True
+
+    def set_low_threshold(self, value):
+        """Sets the sensor low threshold value"""
+        self.low_thresholds[1] = value
+        return True
+
+    def get_high_critical_threshold(self):
+        """Returns the sensor critical high threshold value"""
+        return self.high_thresholds[2]
+
+    def get_low_critical_threshold(self):
+        """Returns the sensor critical low threshold value"""
+        return self.low_thresholds[2]
+
+    def set_high_critical_threshold(self, value):
+        """Sets the sensor critical high threshold value"""
+        self.high_thresholds[2] = value
+        return True
+
+    def set_low_critical_threshold(self, value):
+        """Sets the sensor critical low threshold value"""
+        self.low_thresholds[2] = value
+        return True
+
+    def get_minimum_recorded(self):
+        """Retrieves the minimum recorded sensor measurement"""
+        tmp = self.get_value()
+        if tmp is None:
+            return None
+        if self.minimum_sensor is None or tmp < self.minimum_sensor:
+            self.minimum_sensor = tmp
+        return self.minimum_sensor
+
+    def get_maximum_recorded(self):
+        """Retrieves the maximum recorded sensor measurement"""
+        tmp = self.get_value()
+        if tmp is None:
+            return None
+        if  self.maximum_sensor is None or tmp > self.maximum_sensor:
+            self.maximum_sensor = tmp
+        return self.maximum_sensor
+
+    def get_position_in_parent(self):
+        """Retrieves 1-based relative physical position in parent device"""
+        return self.position
+
+
+class VoltageSensorFs(SensorFs, VoltageSensorBase):
+    """File system based voltage sensor class"""
+
+    DEVICE_TYPE = "voltage_sensor"
+
+    def __init__(self, **kw):
+        super(VoltageSensorFs, self).__init__(self.DEVICE_TYPE, **kw)
+
+    @staticmethod
+    def factory(sensors_data):
```

Comment:
@bmridul Factory class need to be repeated for every Sensor. Can you make it agnostic to the sensor Class so that we don't need to add new factory for future newer sensors?

Reply:
Yes, the difference is sensor type. I will check how to do this.

Reply:
@bmridul here is one way

<img width="306" alt="image" src="https://github.com/sonic-net/sonic-platform-common/assets/45705344/6f0464f9-a583-4330-aa7a-0fdb6bd3c516">

Reply:
Fixed.